sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java

```diff
@@ -186,8 +188,12 @@ public DefaultAzureCredential build() {
     }
 
     private ArrayList<TokenCredential> getCredentialsChain() {
-        ArrayList<TokenCredential> output = new ArrayList<TokenCredential>(7);
+        WorkloadIdentityCredential workloadIdentityCredential = CoreUtils.isNullOrEmpty(managedIdentityClientId) ?  null : getWorkloadIdentityCredentialIfAvailable();
```

Comment:
If `managedIdentityClientId` is a public option, we should have another specifically for workload identity e.g. `workloadIdentityClientId` because we want to distinguish workload identity from managed identity. The precedence would then be:
1. workloadIdentityClientId
2. managedIdentityClientId
3. AZURE_CLIENT_ID

Reply:
added the workloadIdentityClientId option.